internal/bminventory/inventory_test.go

```diff
@@ -480,6 +480,40 @@ var _ = Describe("RegisterHost", func() {
 		Expect(apiErr.StatusCode()).Should(Equal(int32(http.StatusNotFound)))
 	})
 
+	It("register host to a cluster while installation is in progress", func() {
```

Comment:
if you see multiple test that have the a common code you can put them in a context with a `BeforeEach` that create the cluster

Reply:
The scenarios there don't  share that much code, apart from the existing code in ``beforeEach`` and ``afterEach``
Let me know if you think otherwise

Reply:
`register_success` and `host_api_failure` the other two tests below create the same cluster no? 
```
		clusterID := strfmt.UUID(uuid.New().String())
		cluster := common.Cluster{
			Cluster: models.Cluster{
				ID:     &clusterID,
				Status: swag.String(models.ClusterStatusInstalling),
			},
		}
```
So i suggest that ether make a function that create the cluster or  put them under the same context.

Reply:
alright, have a look now